WeatherApp/WeatherHandler.swift

```diff
@@ -71,25 +138,13 @@ class WeatherHandler {
                         let maxTemperature = Measurement(value: maxValue, unit: UnitTemperature.celsius)
                         let weatherData = WeatherData(date: weatherDate, avgTemp: currTemperature, minTemp: minTemperature, maxTemp: maxTemperature, condition: weatherCondition, conditionID: conditionID)
                         
-                        weatherDataList.append(weatherData)
+                        city.weatherData.append(weatherData)
```

Comment:
shouldn't you clear city.weatherData before appending? 
this could easily pile up duplicate weatherData if called multiple times for the same city

Reply:
It seems it dupes data based on your usage of the weatherForCity function

Reply:
I am not sure of the changes you made here, but it seems that at some point you had the correct implementation for this.